### PR TITLE
Generate azure instance types.

### PR DESCRIPTION
**Pull request by nilo19**
generate `azure_instance_types.go` by `go generate`

Fixes: #2327

/kind bug
/area provider/azure

### CODE REVIEW

**Comment by feiskyer** [comment]
Boilerplate header is wrong for: /home/travis/gopath/src/k8s.io/autoscaler/cluster-autoscaler/cloudprovider/azure/azure_instance_types/gen.go

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//go:generate go run ec2_instance_types/gen.go
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go]: ./azure_instance_types/gen.go

File: cluster-autoscaler/cloudprovider/azure/azure_instance_types.go [hunk 2]
```diff
@@ -16,15 +16,16 @@ limitations under the License.
...
+type InstanceType struct {
 	InstanceType string
 	VCPU         int64
 	MemoryMb     int64
 	GPU          int64
 }
 
 // InstanceTypes is a map of azure resources
-var InstanceTypes = map[string]*instanceType{
+var InstanceTypes = map[string]*InstanceType{
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_instance_types.go]: could you update the generated instance types together?

Comment (nilo19) [reply cluster-autoscaler/cloudprovider/azure/azure_instance_types.go]: what do you mean?

Comment (nilo19) [reply cluster-autoscaler/cloudprovider/azure/azure_instance_types.go]: sure

**Comment by nilo19** [comment]
@piosz could you help to review?

**Comment by feiskyer** [comment]
@losipiuk could you help to approve the changes?

**Comment by losipiuk** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**